doc/devel/release_guide.rst

```diff
@@ -6,27 +6,98 @@ Doing a matplotlib release
 
 A guide for developers who are doing a matplotlib release.
 
-* Edit :file:`__init__.py` and bump the version number
+Development (beta/RC) release checklist
+---------------------------------------
+
+- [ ] Testing
+- [ ] Create empty REL: commit
+- [ ] Create tag
+- [ ] Push tags + branches to GH
+- [ ] Notify mac/windows/conda builders
+- [ ] Merge everything to master
+- [ ] announce release
+
+Final release checklist
+-----------------------
+
+- [ ] Testing
+- [ ] update gh states
+- [ ] Create empty REL: commit
+- [ ] Create tag
+- [ ] Create branches
+- [ ] Push tags + branches to GH
+- [ ] release / DOI management
+- [ ] Notify mac/windows/conda builders
+- [ ] Merge everything to master
+- [ ] build documentation
+- [ ] upload to pypi / sf
+- [ ] deploy updated documentation
+- [ ] announce release
+
+Details
+-------
 
 .. _release-testing:
 
 Testing
 =======
 
-* Run all of the regression tests by running ``python tests.py``
-  at the root of the source tree.
+We use `travis-ci <https://travis-ci.org/matplotlib/matplotlib>`__ for
+continuous integration.  When preparing for a release, the final
+tagged commit should be tested locally before it is uploaded::
 
-* Run :file:`unit/memleak_hawaii3.py` and make sure there are no
-  memory leaks
+   python tests.py --processes=8 --process-timeout=300
 
-* try some GUI examples, e.g., :file:`simple_plot.py` with GTKAgg, TkAgg, etc...
+In addition ::
 
-* remove font cache and tex cache from :file:`.matplotlib` and test
-  with and without cache on some example script
+   python unit/memleak_hawaii3.py
```

Comment:
Good point.  I normally work in a conda env so python is python3 (and most of my machines are arch so python is also python3) so I forget that is not universally the case.